server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -553,6 +556,10 @@ public static XContentType parseContentType(List<String> header) {
         throw new IllegalArgumentException("empty Content-Type header");
     }
 
+    public Version getCompatibleVersion() {
+        return compatibleVersion;
+    }
```

Comment:
Instead of exposing the Version directly, can we expose a boolean instead ?

```
boolean applyRestCompatiblity() { 
   return compatibleVersion.major == Version.CURRENT.major - 1;
}
```

Reply:
I think this should be tackled in a separate issue to discuss.
I am worried that if we only have a boolean flag to indicate compatibility we would face problems during release periods.
Let's say we released a version 8, we won't be able to differentiate the code that was meant to be compatible with 7 and the code that was introduced after feature freeze and was meant to be compatible with 8 (talking to ES9)
Having only a flag would mean that work on future (compatible with 8 when the ES8 was just released) compatible code would be impossible until all the "old" (compatible with 7) code is removed. 
I think it would also be prone to be accidentally missed and could be incorrectly considered to be compatible with 8

Reply:
I raised a separate issue to discuss this. https://github.com/elastic/elasticsearch/issues/68100

Reply:
sounds good, we'll continue discussion over there.

Reply:
great, in that case I will merge this PR and a potential change to boolean method would be handled in a separate PR.